src/python/pants/util/help_string.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import inspect
+from dataclasses import dataclass
+from typing import Any, Callable, Mapping, cast
+
+
+def help_string_factory(vars: Mapping[str, Any] | None = None) -> Callable[[str], HelpString]:
+    """Returns a factory function that can be used to create `HelpString` descriptors.
+
+    The `HelpString` objects returned by the factory function can be used to refer to any object
+    that exists in the namespspace provided by `vars`
+
+    When `vars` is `None`, a reference is kept to the global namespace where `help_string_generator()`
+    is called, but the values are only retrieved at evaluation time.
+
+    In practice, this allows `help` literals to refer to properties of global classes that are
+    defined later in the same file, allowing for help strings to refer to actual `alias`es.
+    """
+    _vars: Mapping[str, Any]
+
+    if vars is None:
+        frame: Any = inspect.currentframe()
+        _vars = frame.f_back.f_globals
```

Comment:
Not sure how this updates into future changes.. but might capture more scenarios (not sure if needed though just felt like pointing this one out)

https://github.com/pantsbuild/pants/blob/1767af454873dc26c66e3ef89306f733bb9ddcdd/src/python/pants/engine/internals/rule_visitor.py#L73

Reply:
ah, might not even apply, as this works one a frame, where as the linked one was for functions... my mind made the connection and I posted uncritically :P